### PR TITLE
feat: Add fragmented mp4 Builder and Reader support

### PR DESCRIPTION
**Pull request by gpeacock**
Adds builder.sign_from_fragmented_files()
Adds Reader::from_fragmented_files()
Adds Reader::from_fragment and from_fragment_async
Adds Reader::to_folder -- to extract all resources
Deprecates all manifest_store constructors. Use Reader instead.
Fixes an issue where we were getting duplicate c2pa_rs ClaimGeneratorInfo values. 
Now we no longer add a c2pa_rs ClaimGeneratorInfo but insert a org.cai.c2pa_rs field into the first claim_generator to track the version. This will be more compatible with 2.1 in the future.

### CODE REVIEW

**Comment by scouten-adobe** [comment]
@gpeacock FYI I re-ran the PR title check and it has passed now.

File: sdk/src/builder.rs [hunk 1]
```diff
@@ -972,6 +976,65 @@ impl Builder {
...
+        self.definition.instance_id = format!("xmp:iid:{}", Uuid::new_v4());
+        if self.definition.title.is_none() {
+            if let Some(title) = path.file_name() {
+                self.definition.title = Some(title.to_string_lossy().to_string());
+            }
+        }
+        Ok(())
+    }
+
+    /// Sign a set of fragmented BMFF files for DASH streaming.
```
Comment (mauricefisher64) [inline sdk/src/builder.rs:1007]: You can remove the data comment.  It should be BMFF based.  So HLS for example should work as well.

Comment (mauricefisher64) [reply sdk/src/builder.rs:1007]: We should add a note here that the code does not work if the files already have C2PA data.  That is a known limitation.  I will add a story to support that.

**Review by dyro — APPROVED** [review]
Caught a grammer mistake and there's a question about deprecated version in the deprecated attributes you added.

**Pull request merged**